python/graphstorm/model/node_gnn.py

```diff
@@ -187,18 +189,46 @@ def node_mini_batch_gnn_predict(model, loader, return_proba=True, return_label=F
             input_feats = data.get_node_feats(input_nodes, device)
             blocks = [block.to(device) for block in blocks]
             pred, emb = model.predict(blocks, input_feats, None, input_nodes, return_proba)
-            preds.append(pred.cpu())
-            embs.append(emb.cpu())
+            for ntype, ntype_labels in data.get_labels(seeds).items():
+                if ntype in labels:
+                    labels[ntype].append(ntype_labels.cpu())
+                else:
+                    labels[ntype] = [ntype_labels.cpu()]
+            if isinstance(pred, dict):
```

Comment:
is `pred` always a dict?

Reply:
Yes. Our function `node_mini_batch_gnn_predict` is only used in evaluation, and its output is set to a dictionary by default in this PR. 

For single task node case, then it is a dictionary with length as 1.

Reply:
i try to understand if we need this if condition. if it's always a dict, why do we need to check its type.

Reply:
Oh, I see your point. 

(1) If the prediction is the output of `gnn_encoder` or `decoder`, it is aways a dictionary
(2) However, there is a special case then prediction is the output of `llm_coder` (e.g., one of the test case in `end2end-tests/graphstorm-nc/mgpu_test.sh`), then the output is just a tensor. 

Maybe we keep this `isinstance` check for now for safe. We can remove it after we make sure all modules return is dictionary.